Makefile

```diff
@@ -42,13 +42,18 @@ manager: fmt vet
 
 # Build the clusterctl binary
 clusterctl: fmt vet
-	go build -o bin/clusterctl ./cmd/clusterctl
+	CGO_ENABLED=0 go build -ldflags '-extldflags "-static" -w -s' \
+	-o bin/clusterctl."$${GOOS:-$$(go env GOHOSTOS)}"_"$${GOARCH:-$$(go env GOHOSTARCH)}" \
+	./cmd/clusterctl
+	@cp -f bin/clusterctl."$${GOOS:-$$(go env GOHOSTOS)}"_"$${GOARCH:-$$(go env GOHOSTARCH)}" bin/clusterctl
 
 clusterctl-in-docker:
 	docker run --rm -v $(CWD):/go/src/sigs.k8s.io/cluster-api-provider-vsphere \
 	  -w /go/src/sigs.k8s.io/cluster-api-provider-vsphere \
-	  -e GOOS -e GOHOSTOS golang:1.12 \
-	  go build -o bin/clusterctl ./cmd/clusterctl
+	  -e CGO_ENABLED=0 -e GOOS="$${GOOS:-linux}" -e GOARCH="$${GOARCH:-amd64}" \
+	  golang:1.12 \
+	  go build -ldflags '-extldflags "-static" -w -s' \
+	  -o bin/clusterctl."$${GOOS:-linux}"_"$${GOARCH:-amd64}" ./cmd/clusterctl
```

Comment:
The bin name changing is going to break some docs and blogs

Reply:
Then I'll add the same hack I have in the normal target. We should always include the arch in a Docker output. This is really just fixing that. But I'll add that hack for now.

Reply:
Actually, the clusterctl binaries are on the releases pages now. It should be grab them from there. If a person wants to build it locally then they should be fine handling this IMO.

Reply:
Fixed for now. We should change this later. This target shouldn't be in the Makefile IMO.